tests/test_db_mysql.py

```
@@ -0,0 +1,148 @@
+        single_result = {"id": 1, "lemma": "test"}
+        mock_db_connection[0].cursor.fetchone.return_value = single_result
+
+        with patch("db_mysql._fetch_all", return_value=(single_result, 0.3)):
+            result, exec_time = fetch_all("SELECT * FROM test", fetch_one=True)
+
+            assert isinstance(result, dict)
+            assert result["id"] == 1
+
+    def test_fetch_all_with_empty_params(self):
+        """Test that None params are converted to empty tuple"""
+        mock_db_connection[0].cursor.fetchone.return_value = {"total": 0}
```

Ruff (0.15.12)

[error] Undefined name `mock_db_connection`

(F821)